src/ExpressionEvaluator/Core/Source/ExpressionCompiler/EvaluationContextBase.cs

```diff
@@ -110,6 +110,8 @@ internal string GetErrorMessageAndMissingAssemblyIdentities(DiagnosticBag diagno
 
         internal abstract ImmutableArray<AssemblyIdentity> GetMissingAssemblyIdentities(Diagnostic diagnostic);
 
+        internal abstract bool HasReferenceToAny(ImmutableArray<AssemblyIdentity> identities);
```

Comment:
Perhaps expose an abstract IEnumerable<AssemblyIdentity> References property and simply loop over that in ExpressCompiler.